zio-http/src/main/scala/zio/http/model/headers/values/Warning.scala

```diff
@@ -0,0 +1,128 @@
+package zio.http.model.headers.values
+
+import java.time.{ZoneId, ZonedDateTime}
+import java.time.format.DateTimeFormatter
+import scala.util.Try
+
+sealed trait Warning {}
+
+/*
+The Warning HTTP header contains information about possible problems with the status of the message.
+  More than one Warning header may appear in a response.
+
+Warning header fields can, in general, be applied to any message.
+  However, some warn-codes are specific to caches and can only be applied to response messages.
+ */
+
+object Warning {
+
+  /*
+     A warning has the following syntax: <warn-code> <warn-agent> <warn-text> [<warn-date>]
+   */
+  final case class WarningValue(code: Int, agent: String, text: String, date: Option[ZonedDateTime] = None)
+      extends Warning
+
+  case object InvalidWarning extends Warning
+
+  def toWarning(warningString: String): Warning = {
+    /*
+      <warn-code>
+       A three-digit warning number.
+         The first digit indicates whether the Warning is required to be deleted from a stored response after validation.
+
+       1xx warn-codes describe the freshness or validation status of the response and will be deleted by a cache after deletion.
+
+        2xx warn-codes describe some aspect of the representation that is not rectified by a validation and
+           will not be deleted by a cache after validation unless a full response is sent.
+     */
+    val warnCode: Int = Try { Integer.parseInt(warningString.split(" ")(0)) }.getOrElse(-1)
+
+    /*
+       <warn-agent>
+         The name or pseudonym of the server or software adding the Warning header (might be "-" when the agent is unknown).
+     */
+    val warnAgent: String = warningString.split(" ")(1)
+
+    /*
+       <warn-text>
+       An advisory text describing the error.
+     */
+    val descriptionStartIndex = warningString.indexOf('\"')
+    val descriptionEndIndex   = warningString.indexOf("\"", warningString.indexOf("\"") + 1)
+    val description           =
+      Try { warningString.substring(descriptionStartIndex, descriptionEndIndex + 1) }.getOrElse("")
+
+    /*
+    <warn-date>
+    A date. This is optional. If more than one Warning header is sent, include a date that matches the Date header.
+     */
+    val dateStartIndex = warningString.indexOf("\"", descriptionEndIndex + 1)
+    val dateEndIndex   = warningString.indexOf("\"", dateStartIndex + 1)
+    val warningDate    = Try {
+      val selectedDate   = warningString.substring(dateStartIndex + 1, dateEndIndex)
+      val expectedFormat = DateTimeFormatter.ofPattern("EEE, dd MMM yyyy HH:mm:ss zzz")
```

Comment:
You can improve performance by pushing constants like this one (especially this one, since it's expensive to compute) into the companion object and making it private.